src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -323,26 +323,44 @@ private static byte[] AppendEndOfLine(ReadOnlySpan<byte> span, bool lineFeedOnly
     private int ParseMultiSpanHeader(TRequestHandler handler, ref SequenceReader<byte> reader)
     {
         var currentSlice = reader.UnreadSequence;
-        var lineEndPosition = currentSlice.PositionOfAny(ByteCR, ByteLF);
 
-        if (lineEndPosition == null)
+        SequencePosition position = currentSlice.Start;
+        // Skip the first segment as the caller already searched it for CR/LF
+        if (!currentSlice.TryGet(ref position, out ReadOnlyMemory<byte> memory) || position.GetObject() == null)
```

Comment:
`currentSlice.TryGet(ref position, out ReadOnlyMemory<byte> memory)` can never return false, right? Can we assert this?